pytorch_lightning/utilities/auto_restart.py

```diff
@@ -570,3 +563,13 @@ def reload_dataloader_state_dict(dataloader: DataLoader, state_dict: Dict[str, A
 
     else:
         raise MisconfigurationException("This shouldn't happen. Please, open an issue on PyTorch Lightning Github.")
+
+
+def _rotate_worker_indices(state, latest_worker_id: int, num_workers: int):
+    """This function is used to rotate the worker indices based on the `latest_worker_id` the training failed
+    on."""
+    if num_workers == 0:
+        return state
```

Comment:
let's also handle the error when `latest_worker_id > num_workers - 1` ?

Reply:
Done !